cql3/statements/select_statement.cc

```diff
@@ -1462,6 +1571,21 @@ std::unique_ptr<prepared_statement> select_statement::prepare(database& db, cql_
                 prepare_limit(db, ctx, _per_partition_limit),
                 stats,
                 std::move(prepared_attrs));
+    } else if (selection->is_aggregate() && selection->is_count() && !restrictions->need_filtering()) {
```

Comment:
I'd prefer the checks to be on the expressions rather than selection, but we can sort that out later.